model/workflow.go

```diff
@@ -70,20 +70,49 @@ func continueAsStructLevelValidation(structLevel validator.StructLevel) {
 	}
 }
 
+// WorkflowStructLevelValidation custom validator
+func workflowStructLevelValidation(structLevel validator.StructLevel) {
+	// unique name of the auth methods
+	// NOTE: we cannot add the custom validation of auth to AuthArray
+	// because `RegisterStructValidation` only works with struct type
+	wf := structLevel.Current().Interface().(Workflow)
+	dict := map[string]bool{}
+
+	for _, a := range wf.BaseWorkflow.Auth {
+		if !dict[a.Name] {
+			dict[a.Name] = true
+		} else {
+			structLevel.ReportError(reflect.ValueOf(a.Name), "[]Auth.Name", "name", "reqnameunique", "")
+		}
+	}
+
+	// start state name exists in states list
+	startExists := false
+	for _, state := range wf.States {
+		if state.Name == wf.BaseWorkflow.Start.StateName {
+			startExists = true
+			break
+		}
+	}
+	if !startExists {
+		structLevel.ReportError(reflect.ValueOf(wf.BaseWorkflow.Start.StateName), "Start", "start", "startnotexists", "")
+	}
+}
+
 // BaseWorkflow describes the partial Workflow definition that does not rely on generic interfaces
 // to make it easy for custom unmarshalers implementations to unmarshal the common data structure.
 type BaseWorkflow struct {
 	// Workflow unique identifier
-	ID string `json:"id" validate:"omitempty,min=1"`
+	ID string `json:"id,omitempty" validate:"required_without=Key"`
 	// Key Domain-specific workflow identifier
-	Key string `json:"key,omitempty" validate:"omitempty,min=1"`
+	Key string `json:"key,omitempty" validate:"required_without=ID"`
 	// Workflow name
 	Name string `json:"name,omitempty"`
 	// Workflow description
 	Description string `json:"description,omitempty"`
 	// Workflow version
 	Version string `json:"version" validate:"omitempty,min=1"`
-	Start   *Start `json:"start,omitempty"`
+	Start   *Start `json:"start,omitempty" validate:"required"`
```

Comment:
I think it's important to populate the BaseWorkflow.Start even if it wasn't defined in json/yaml. Because the application using the SDK knows precisely what the start is.

I added 'required' in Workflow.Start as a second form of a check that, at some point, start was defined.

Also, that rule in the specification // if the start is not defined, use the first state.

https://github.com/serverlessworkflow/sdk-go/pull/156/files/49add33759f3b194d9f4060a8fd3efdf6900b8b6#diff-6a5d032df78474ceeda4fabb343d1ee198295867c5339285f03e7e65459f5985R215-R221